colossalai/inference/server/api_server.py

```diff
@@ -184,13 +194,21 @@ def parse_args():
 if __name__ == "__main__":
     args = parse_args()
     inference_config = InferenceConfig.from_dict(vars(args))
-    model = AutoModelForCausalLM.from_pretrained(args.model)
     tokenizer = AutoTokenizer.from_pretrained(args.model)
+    colossalai_backend_port = find_available_ports(1)[0]
```

Comment:
![image](https://github.com/hpcaitech/ColossalAI/assets/26091437/9767755f-124c-4fae-aed8-b2ea5253e5ba)
Consider adding the above logic inside the engine to avoid initiating TP logic even when using a single GPU.

Reply:
maybe this be left for you to change